### PR TITLE
docs: Add package manager dropdown on copy button

### LINKED ISSUE #642: Package managers dropdown on code copy
### Describe the feature in detail (code, mocks, or screenshots encouraged)

[ui.shadcn.com](https://ui.shadcn.com) has a feature that I like: when you press the right button to copy a command in the docs, you get the choice to choose the package manager you want to copy for.

This feature doesn't exist on [shadcn-svelte.com](https://shadcn-svelte.com). Are you okay for me to implement it (using the dropdown menu component?), making it as close as possible to the original design from shadcn?

### What type of pull request would this be?

Enhancement

### Provide relevant links or additional information.

_No response_

### PR DESCRIPTION
**Pull request by WarningImHack3r**
### Before submitting the PR, please make sure you do the following

- [x] If your PR isn't addressing a small fix (like a typo), it references an issue where it is discussed ahead of time and assigned to you. In many cases, features are absent for a reason.
- [x] Prefix your PR title with `feat:`, `fix:`, `chore:`, or `docs:`.
- [x] This message body should clearly illustrate what problems it solves.
- [x] Follows the [contribution guidelines](https://github.com/huntabyte/shadcn-svelte/blob/main/CONTRIBUTING.md).
- [x] Format & lint the code with `pnpm format` and `pnpm lint`

Fixes #642.

I finally got time to lean into this feature. What took time was getting more familiar with mdsvex and rehype/remark plugins, which I never really did before (only using `md`➜`html`, never contributed to them, or even wondered how they were working).

At the end of the day, instead of blindly copying what shadcn did [for the original website](https://github.com/shadcn-ui/ui/blob/main/apps/www/lib/rehype-npm-command.ts), I preferred a Sveltier approach: simpler, type-safe and shorter.

Hope you like it!

> I also added some comments in the mdsvex config file as I was messing around, I can totally remove that if you think it's unnecessary/not the point of the PR

### CODE REVIEW

File: apps/www/mdsvex.config.js [hunk 1]
```diff
@@ -181,13 +195,10 @@ function rehypeComponentPreToPre() {
 			if (node?.type === "element" && node?.tagName === "Components.pre") {
 				node.tagName = "pre";
 			}
-
-			if (node?.type === "element" && node?.tagName === "pre") {
-				//
-			}
 		});
```
Comment (huntabyte) [inline apps/www/mdsvex.config.js:198]: Thanks for the cleanup!

File: apps/www/src/lib/components/docs/copy-button.svelte [hunk 2]
```diff
@@ -2,12 +2,52 @@
...
+			};
+		}
+
+		// npx create
+		else if (value.startsWith("npx create-")) {
+			commands = {
+				npm: value,
+				yarn: value.replace("npx create-", "yarn create "),
+				pnpm: value.replace("npx create-", "pnpm create "),
+				bun: value.replace("npx create-", "bunx --bun")
```
Comment (huntabyte) [inline apps/www/src/lib/components/docs/copy-button.svelte:36]:
```suggestion
				yarn: value.replace("npx create ", "yarn create "),
				pnpm: value.replace("npx create ", "pnpm create "),
				bun: value.replace("npx create ", "bunx --bun")
```

I'm not seeing anywhere that a `-` follows create, so I think this is causing the `npm create svelte@latest my-app` block to not work like the others!

**Review by huntabyte — CHANGES_REQUESTED** [review]
This is dope! Thanks a ton, just a minor thing! Let me know if you want me to take care of it (I try not to intrude on PRs without permission!)

Also, the comments are much appreciated! Probably should have been there from the beginning! The plugins are quite the work of art right? 😂 Took me a bit to figure out how to get this site to behave like the original!

File: apps/www/src/lib/components/docs/copy-button.svelte (hunk 2 above)
Comment (WarningImHack3r) [reply apps/www/src/lib/components/docs/copy-button.svelte:36]: I honestly copied the code snippet I linked in the PR description and didn't change it much 😅 I just added the `yarn dlx` command and put `else`s between conditions and that's it!

That being said, it's a good observation! However, as both can apparently exist (see [npm docs](https://docs.npmjs.com/cli/v10/commands/npm-init)), I'm in favor of merging both options!

**Comment by WarningImHack3r** [comment]
> This is dope! Thanks a ton, just a minor thing! Let me know if you want me to take care of it (I try not to intrude on PRs without permission!)

Haha yeah you can modify it no problem!!

> The plugins are quite the work of art right? 😂 Took me a bit to figure out how to get this site to behave like the original!

OMG yeah, fortunately I had some basis, and my recent work on [svelte-changelog](https://github.com/WarningImHack3r/svelte-changelog/commit/083f47518f34429336f418931edd1cbebbd4ebdf) got me in better shape to work on that! When I got assigned this feature and that I later noticed it was not as simple (I thought) as a few more lines of Svelte I was like man, I should've really planned my feature before having to dig into that for the first time! I wasn't even sure I could bring the feature I requested into life and I'd be looking like a jerk haha.  
I saw your tweet about a lack of better solutions nowadays in the Svelte ecosystem (like Docausaurus?) and I agree so bad!

**Comment by huntabyte** [comment]
https://github.com/huntabyte/shadcn-svelte/assets/64506580/98ddadca-1f30-49bf-8377-e9f81faa9cf6

The dropdown doesn't appear to work on this one command 🤔

**Comment by WarningImHack3r** [comment]
Indeed! It isn't in the conditions we discussed, because it wasn't needed for React. Adding that case right now, sorry.

**Review by huntabyte — APPROVED** [review]

**Pull request merged**